src/Transitions/Collapse.js

```diff
@@ -0,0 +1,94 @@
+/*eslint-disable react/prop-types */
+'use strict';
+import React from 'react';
+import Transition from './Transition';
+import domUtils from '../utils/domUtils';
+import createChainedFunction from '../utils/createChainedFunction';
+
+let capitalize = str => str[0].toUpperCase() + str.substr(1);
+
+// reading a dimension prop will cause the browser to recalculate,
+// which will let our animations work
+let triggerBrowserReflow = node => node.offsetHeight; //eslint-disable-line no-unused-expressions
+
+const MARGINS = {
+  height: ['marginTop', 'marginBottom'],
+  width:  ['marginLeft', 'marginRight']
+};
+
+function getDimensionValue(dimension, elem){
+  let value = elem[`offset${capitalize(dimension)}`];
+  let computedStyles = domUtils.getComputedStyles(elem);
+  let margins = MARGINS[dimension];
+
+  return (value +
+    parseInt(computedStyles[margins[0]], 10) +
+    parseInt(computedStyles[margins[1]], 10)
+  );
+}
+
+class Collapse extends React.Component {
+
+  constructor(props, context){
+    super(props, context);
+  }
+
+  render() {
+    let entering = createChainedFunction(this.handleEntering.bind(this), this.props.onEntering);
+    let entered = createChainedFunction(this.handleEntered.bind(this), this.props.onEntered);
+    let exit = createChainedFunction(this.handleExit.bind(this), this.props.onExit);
```

Comment:
If this could break pure render then this should be in the constructor right? Since this is a HoC for other components. I'd rather nip future bugs with pure render in the but now if we can.